cli/src/semgrep/core_runner.py

```diff
@@ -269,10 +270,21 @@ async def _core_stdout_processor(self, stream: asyncio.StreamReader) -> None:
                 break
 
             if line_bytes == b".\n" and not reading_json:
-                num_scanned_targets += 1
+                # We expect to see 3 dots for each target, when running interfile analysis:
+                # - once when finishing phase 4, name resolution, on that target
+                # - once when finishing phase 5, taint configs, on that target
+                # - once when finishing analysis on that target as usual
+                #
+                # However, for regular OSS Semgrep, we only print one dot per
+                # target, that being the last bullet point listed above.
+                #
+                # So a dot counts as 0.33 files progress if running Pro, but 1
+                # file's progress if running the OSS engine.
+                advanced_targets = 1 / 3 if self._engine_type.is_interfile else 1
```

Comment:
<img width="358" alt="CleanShot 2023-03-01 at 12 06 18@2x" src="https://user-images.githubusercontent.com/1060436/222211062-a940e2a7-dc92-449c-9889-c0b21ebed487.png">

the floating point drift from incrementing by 1/3 is making me uncomfortable; could we just always use integers? set the total to three times the current value, and advance by 3 when not interfile

Reply:
that's a fair point, fixed!